### ISSUE TITLE
Ignore sections with just comments and empty lines

### ISSUE THREAD

**Issue by gianlucam76** [projectsveltos/sveltos-manager#176]
Referenced ConfigMaps and Secrets contains YAML of one ore more Kubernetes resources that need to deployed in cluster matching ClusterProfile's clusterSelector.

Currently if one of those section is just comments and empty lines, sveltos fails.

For instance if following section is contained in ConfigMap.Data
```
---
# This file is generated from the individual YAML files by generate-provisioner-deployment.sh. Do not
# edit this file directly but instead edit the source files and re-render.
#
# Generated from:
#       examples/contour/01-crds.yaml
#       examples/gateway/00-crds.yaml
#       examples/gateway/00-namespace.yaml
#       examples/gateway/01-admission_webhook.yaml
#       examples/gateway/02-certificate_config.yaml
#       examples/gateway-provisioner/00-common.yaml
#       examples/gateway-provisioner/01-roles.yaml
#       examples/gateway-provisioner/02-rolebindings.yaml
#       examples/gateway-provisioner/03-gateway-provisioner.yaml

---
```

sveltos will fail.

Sveltos should skip sections that contain only comments or empty lines. 

The only method that needs to change is

```
func collectContent(ctx context.Context, clusterSummary *configv1alpha1.ClusterSummary,
	data map[string]string, logger logr.Logger) ([]*unstructured.Unstructured, error) {
```

where data is essentially the content of ConfigMap.Data or decoded Secret.Data.

**Comment by galiev** [comment]
Sorry, could you help me please. Skip comments should work for helm addons?
I use projectsveltos/addon-controller-amd64:v0.13.0 and try install cilium v1.13.4:
```bash
E0720 14:07:34.549044       1 handlers_helm.go:1457] "deployer: failed to get policy from Data # Source: cilium/templates/cilium-secrets-namespace.yaml\n# Only create the namespace if it's differe" err="failed to decode k8s resource # Source: cilium/templates/cilium-secrets-namespac. Err: Object 'Kind' is missing in 'null'" worker="15" key="ash:::ash-cluster:::Capi:::cilium-capi-ash-cluster:::Helm:::false" cluster="ash/ash-cluster" clusterSummary="cilium-capi-ash-cluster" admin="/" releaseNamespace="kube-system" releaseName="cilium" release="cilium" releaseNamespace="kube-system" chart="cilium/cilium" chartVersion="1.13.4"
I0720 14:07:34.549080       1 handlers_helm.go:1500] "deployer: failed to collect helm resources failed to decode k8s resource # Source: cilium/templates/cilium-secrets-namespac. Err: Object 'Kind' is missing in 'null'" worker="15" key="ash:::ash-cluster:::Capi:::cilium-capi-ash-cluster:::Helm:::false" cluster="ash/ash-cluster" clusterSummary="cilium-capi-ash-cluster" admin="/" releaseNamespace="kube-system" releaseName="cilium" release="cilium" releaseNamespace="kube-system" chart="cilium/cilium" chartVersion="1.13.4"
I0720 14:07:34.549168       1 worker.go:238] "deployer: added to result with err failed to decode k8s resource # Source: cilium/templates/cilium-secrets-namespac. Err: Object 'Kind' is missing in 'null'" worker="15" key="ash:::ash-cluster:::Capi:::cilium-capi-ash-cluster:::Helm:::false"
```
[cilium-secrets-namespace.yaml](https://github.com/cilium/cilium/blob/v1.13.4/install/kubernetes/cilium/templates/cilium-secrets-namespace.yaml):
```yaml
{{- if and .Values.ingressController.enabled .Values.ingressController.secretsNamespace.create .Values.ingressController.secretsNamespace.name }}
---
apiVersion: v1
kind: Namespace
metadata:
  name: {{ .Values.ingressController.secretsNamespace.name | quote }}
{{- end}}

# Only create the namespace if it's different from Ingress secret namespace or Ingress is not enabled.
{{- if and .Values.gatewayAPI.enabled .Values.gatewayAPI.secretsNamespace.create .Values.gatewayAPI.secretsNamespace.name
  (or (not (and .Values.ingressController.enabled .Values.ingressController.secretsNamespace.create .Values.ingressController.secretsNamespace.name))
      (ne .Values.gatewayAPI.secretsNamespace.name .Values.ingressController.secretsNamespace.name)) }}
---
apiVersion: v1
kind: Namespace
metadata:
  name: {{ .Values.gatewayAPI.secretsNamespace.name | quote }}
{{- end}}
```

**Comment by galiev** [comment]
@gianlucam76 , thank you!
```yaml
apiVersion: config.projectsveltos.io/v1alpha1
kind: ClusterProfile
metadata:
  name: cilium
spec:
  clusterSelector: cni=cilium
  syncMode: Continuous
  helmCharts:
    - repositoryURL:    https://helm.cilium.io/
      repositoryName:   cilium
      chartName:        cilium/cilium
      chartVersion:     1.13.4
      releaseName:      cilium
      releaseNamespace: kube-system
      helmChartAction:  Install
      values: |
        ipam:
          mode: kubernetes
        kubeProxyReplacement: strict
        securityContext:
          capabilities:
            ciliumAgent:
              - CHOWN
              - KILL
              - NET_ADMIN
              - NET_RAW
              - IPC_LOCK
              - SYS_ADMIN
              - SYS_RESOURCE
              - DAC_OVERRIDE
              - FOWNER,SETGID
              - SETUID
            cleanCiliumState:
              - NET_ADMIN
              - SYS_ADMIN
              - SYS_RESOURCE
        cgroup:
          autoMount:
            enabled: false
          hostRoot: /sys/fs/cgroup
        hostFirewall:
          enabled: true
        policyEnforcementMode: default
        policyAuditMode: true
        k8s:
          requireIPv4PodCIDR: true
          requireIPv6PodCIDR: true
        ipv4:
          enabled: true
        ipv6:
          enabled: true
        hostPort:
          enabled: true
        nodePort:
          enabled: true
        k8sServiceHost: {{ .Cluster.spec.controlPlaneEndpoint.host }}
        k8sServicePort: {{ .Cluster.spec.controlPlaneEndpoint.port }}
     ```

**Comment by gianlucam76** [comment]
@galiev 

I was able to repro with your config. 

"status:
    featureSummaries:
    - failureMessage: 'failed to decode k8s resource # Source: cilium/templates/cilium-secrets-namespac.
        Err: Object ''Kind'' is missing in ''null'''
      featureID: Helm
"

And I have a fix for it that worked 

" - featureID: Helm
      hash: RLOYa9fYQqkdeGuOkBPFa8SE6c+6cUc9ZL793Xj1S9Y=
      lastAppliedTime: "2023-07-20T15:00:54Z"
      status: Provisioned
"

It is same problem described in this issue. It is just I fixed for referenced Secret/ConfigMap but did not fix for Helm charts (those are two different paths).

Fix was simple, just call the removeCommentsAndEmptyLines in collectHelmContent same as already done for collectContent from ConfigMap/Secret. 

Fix is here: https://github.com/projectsveltos/addon-controller/pull/283/files

**Comment by gianlucam76** [comment]
@galiev  If you want to be unblocked, please replace the image used for addon-controller manager (projectsveltos/addon-controller-amd64:v0.13.0) with projectsveltos/addon-controller-amd64:dev-fix

thank you again for reporting the issue.

**Comment by galiev** [comment]
@gianlucam76 thank you very much!

**Comment by gianlucam76** [comment]
@galiev there was another kind of major fix needed, so I ended up releasing v0.14.0 which is more stable release.

Thank you for reporting this issue. And in future if you hit any other issue (hopefully not) or you see a feature which would be good to have please let me know.

Thank you again!
